transport/outbound.go

```diff
@@ -27,6 +27,12 @@ import "golang.org/x/net/context"
 // Outbound is a transport that knows how to send requests for procedure
 // calls.
 type Outbound interface {
+	// Sets up the outbound to start making calls.
```

Comment:
I wouldn't worry about the synchronization overhead - instead, making it dummy-proof for new users should probably be the goal. If `Start` *must* be called before `Call`, then we should enforce that if it's possible.